cyprecice/cyprecice.pyx

```diff
@@ -1211,6 +1211,223 @@ cdef class Interface:
         self.thisptr.readScalarData (data_id, vertex_id, _value)
         return _value
 
+    def write_block_vector_gradient_data (self, data_id, vertex_ids, gradientValues):
+        """
+        Writes vector gradient data given as block. This function writes values of specified vertices to a dataID.
+        Values are provided as a block of continuous memory. Values are stored in a numpy array [N x D] where N = number
+        of vertices and D = dimensions of geometry squared.
+
+        Parameters
+        ----------
+        data_id : int
+            Data ID to write to.
+        vertex_ids : array_like
+            Indices of the vertices.
+        gradientValues : array_like
+             Gradient values differentiated in the spacial direction (dx, dy) for 2D space, (dx, dy, dz) for 3D space
+
+        Notes
+        -----
+        Previous calls:
+            Count of available elements at values matches the configured dimension
```

Comment:
Is this a copy-left artifact? Which values are you referring to? Do you mean gradients here?

Reply:
Yes it should be `Count of available elements at values matches the number of gradient components * size` right?

Reply:
In case you write this, I still don't know which values you are referring to. There are no values mentioned anywhere.

Reply:
I think the same here in C++ doc
https://github.com/precice/precice/blob/d83d3c789172acca414233bd65524297c4ce3438/src/precice/SolverInterface.hpp#L1165 
 
 https://github.com/precice/python-bindings/blob/ed5de3dbb6fd29faf1421c2c009db3bf1d5cf203/cyprecice/cyprecice.pyx#L1217-L1218
 
Here it mentioned `[NxD]` so I think `Count of available elements at values matches the configured dimension` was enough or more explicitly we can write `Count of available elements at values matches the number of gradient components`